### DISCUSSION TITLE
libsodium weirdness with x64-mingw-dynamic

### DISCUSSION THREAD

**Discussion by HunterZ** [microsoft/vcpkg#41224] [General]
Trying to build a project using the x64-mingw-dynamic triplet that has a second-order libsodium dependency.

For some reason vcpkg downloads an entire separate set of MSYS2 packages even though I'm already running from an MSYS2 MinGW64 environment.

It then builds for 40+ minutes before triggering some Windows Defender threat warnings and finally issuing an error about not being able to rename a README file.

What is this monstrosity?!

**Comment by dg0yt** [comment]
> For some reason vcpkg downloads an entire separate set of MSYS2 packages even though I'm already running from an MSYS2 MinGW64 environment.

All windows builds use the same set of msys2 tools. Regardless of the mingw compiler.

> It then builds for 40+ minutes before triggering some Windows Defender threat warnings 

This happens on Windows with various Antivirus products, occasionally. Regardless of the mingw compiler.

> finally issuing an error about not being able to rename a README file.

Might be a bug, but the information is not sufficient.
